### DISCUSSION TITLE
Why choose javalin instead of spring boot as the web framework

### DISCUSSION THREAD

**Discussion by JinsYin** [apache/amoro#1218] [Q&A]
Why choose javalin instead of spring boot as the web framework

**Comment by Aireed** [comment]
Javalin is a lightweigth web framework.  As a open-source project, it has 6.4K star and five hundred Forks.  It's used by many projects.

**Reply by JinsYin** [reply]
Do you consider performance and other factors? I am planning to use spring-boot refactoring. Is this feasible?

**Reply by Aireed** [reply]
> Do you consider performance and other factors? I am planning to use spring-boot refactoring. Is this feasible?

Sorry for the late reply.  yes, performance is the first important factor when i choosed the framework.  
 From the perspective of product functionality, web requests and open api requests are not too heavy, Mainly requests from thrift services.  
Please let me know if  you have encounted some performance problem?

**Reply by JinsYin** [reply]
Thanks for your reply. I have analyzed some other metadata services, such as [metacat](https://github.com/Netflix/metacat/blob/master/metacat-main/build.gradle#L49-L54), which are also using spring boot to provide REST API services. Therefore, I will also plan to use spring boot instead of javaling and take benchmark test.

**Reply by Aireed** [reply]
If there is anything we can do to help, please feel free to communicate.

**Reply by JinsYin** [reply]
Okay, Thanks.

**Comment by shidayang** [comment]
I think spring-boot is better, IOC,DI,gloabl exception handler, mybatis plugin ..... is very useful
